ooni/reporter.py

```diff
@@ -313,162 +363,176 @@ def finish(self):
-                'created_at': datetime.now(),
-                'status': 'created',
-                'collector': collector_settings,
-                'report_id': report_id
-            }
-        return report_id
-
-    def created(self, report_file, collector_settings, report_id):
-        return self.run(self._created, report_file,
-                        collector_settings, report_id)
-
-    def _creation_failed(self, report_file, collector_settings):
-        with self.edit_log() as report:
-            report[report_file] = {
-                'pid': os.getpid(),
-                'created_at': datetime.now(),
-                'status': 'creation-failed',
-                'collector': collector_settings
-            }
+    @defer.inlineCallbacks
+    def update_log(self, measurement_id, value):
+        lock = self._lock_for_report_log(measurement_id)
+        yield lock.deferUntilLocked()
 
-    def creation_failed(self, report_file, collector_settings):
-        return self.run(self._creation_failed, report_file,
-                        collector_settings)
+        report_log_file = self._get_report_log_file(measurement_id)
+        with report_log_file.open('w+') as out_file:
+            entry = value
+            entry['last_update'] = datetime.utcnow().strftime(self._date_format)
+            json.dump(entry, out_file)
 
-    def _incomplete(self, report_file):
-        with self.edit_log() as report:
-            if report[report_file]['status'] != "created":
-                raise errors.ReportNotCreated()
-            report[report_file]['status'] = 'incomplete'
+        lock.unlock()
 
-    def incomplete(self, report_file):
-        return self.run(self._incomplete, report_file)
+    @defer.inlineCallbacks
+    def remove_log(self, measurement_id):
+        lock = self._lock_for_report_log(measurement_id)
+        yield lock.deferUntilLocked()
 
-    def _closed(self, report_file):
-        with self.edit_log() as report:
-            rs = report[report_file]['status']
-            if  rs != "created" and rs != "incomplete":
-                raise errors.ReportNotCreated()
-            del report[report_file]
+        report_log_file = self._get_report_log_file(measurement_id)
+        try:
+            log.debug("Deleting log file")
+            report_log_file.remove()
+        except Exception as exc:
+            log.exception(exc)
 
-    def closed(self, report_file):
-        return self.run(self._closed, report_file)
+        lock.unlock()
+
+    @defer.inlineCallbacks
+    def get_incomplete(self):
+        incomplete_reports = []
+        all_entries = yield self.get_report_log_entries()
+        for entry in all_entries[:]:
+            # This means that the measurement itself is incomplete
+            if entry['completed'] is False:
+                continue
+            if entry['status'] in ('created',):
+                try:
+                    os.kill(entry['pid'], 0)
+                except OSError:
```

Comment:
Maybe also here you would like to distinguish between different reasons why `kill()` failed?